hedera-node/src/main/java/com/hedera/services/fees/charging/NarratedLedgerCharging.java

```diff
@@ -224,4 +227,22 @@ private void initEffPayerBalance(EntityNum effPayerId) {
 		}
 		effPayerStartingBalance = payerAccount.getBalance();
 	}
+
+	private void adjustFundingAndStakingBalances(final long totalFee) {
+		final var stakingRewardFee = calculateStakingRewardFee(totalFee);
+		final var nodeRewardFee = calculateNodeRewardFee(totalFee);
+		final var fundingAccountFee = totalFee - stakingRewardFee - nodeRewardFee;
+
+		ledger.adjustBalance(dynamicProperties.fundingAccount(), fundingAccountFee);
+		ledger.adjustBalance(STAKING_REWARD_FUND_ACCOUNT.toGrpcAccountId(), stakingRewardFee);
+		ledger.adjustBalance(NODE_REWARD_FUND_ACCOUNT.toGrpcAccountId(), nodeRewardFee);
+	}
+
+	private long calculateStakingRewardFee(long totalFee) {
+		return (long) ((dynamicProperties.getStakingRewardPercent() * totalFee) / 100.0);
```

Comment:
We can just use integer arithmetic here (and below); no need to introduce a `double` imo.

Reply:
Changed to divide by 100